src/inactive-blog-filter.js

```diff
@@ -0,0 +1,143 @@
+ * @returns {boolean} - bool value
+ */
+function isRedlisted(feedUrl, redItem) {
+  return (feedUrl === redItem.url);
+}
+
+/**
+ * Simple function that takes the difference between current date and post date
+ * If difference is greater than threshold, the blog is considered inactive
+ * @param {Date} postDate - javascript date object of the post being checked
+ * @returns {number} - date difference in milliseconds
+ */
+function dateDiff(postDate) {
+  const currentDate = new Date();
+  return currentDate - postDate;
+}
+
+/**
+* Callback for redlist check
+* @callback checkCallback
+* @param {boolean} result - true/false whether or not a feed url is redlisted
+*/
+
+/**
+ * Checks if feed url is redlisted
+ * @param {string} feedUrl - url of the feed to check against redlist
+ * @param {checkCallback} callback - a callback that runs after the check
+ */
+function check(feedUrl, callback) {
+  // Read redlist file
+  fs.readFile('feeds-redlist.json', (err, redListRaw) => {
+    if (err) {
+      // Error reading file
+      console.warn('problem reading redlist file:', err.message);
+      callback(false);
+      return;
+    }
+
+    if (redListRaw.length === 0) {
+      // File is empty
+      console.warn('redlist file is empty, no filters applied\n\tsee: feeds-redlist.json');
+      callback(false);
+      return;
+    }
+
+    // Concat to array no matter what
+    const redList = [].concat(JSON.parse(redListRaw));
+
+    callback(redList.some(isRedlisted.bind(null, feedUrl)));
+  });
+}
+
+/*
+ * Performs a separate sweep of all feeds to see which ones are inactive,
+ * then updates last post date in feeds-redlist.json
+ * *
+ * Integration with system can be improved once feed-worker is augmented to
+ * pass feed data more intuitively
+ * *
+ * Due to amount of operations, this can be run periodically instead of with every feed update
+ */
+function update() {
+  // Read the feeds list file
+  fs.readFile('feeds.txt', 'utf8', (err, lines) => {
+    if (err) {
+      console.error('unable to read initial list of feeds, cannot update', err.message);
+      return;
+    }
+
+    // Divide the file into separate lines
+    const feedUrlList = lines
+      .split(/\r?\n/)
+      // Basic filtering to remove any ines that don't look like a feed URL
+      .filter((line) => line.startsWith('http'))
+      // Convert this into an Object of the form expected by our queue
+      .map((url) => ({ url }));
+
+    const redlistUpdate = [];
+    let linesRead = 0;
+
+    feedUrlList.forEach(async (feedItem) => {
+      const feed = await feedParser(feedItem);
+      const feedUrl = feedItem.url;
+
+      let recentPostDate = new Date();
+
+      // In case of invalid/ dead feeds
+      if (typeof feed[0] !== 'undefined') {
+        recentPostDate = new Date(feed[0].date);
+
+        // Check if the blog is inactive
+        // !!!  Older than 1 year = inactive, but we can change this or
+        // !!!  parameterize it in a config somewhere
```

Comment:
We have a config in `.env`.  You need to add something to the `env.sample` file declaring this with a named environment variable, and read it in from `process.env`

Reply:
Oops, didn't notice this! I'll address it asap.

Reply:
You should [watch the repo](https://help.github.com/en/github/receiving-notifications-about-activity-on-github/watching-and-unwatching-repositories) so you can follow what's happening.